### PR TITLE
Documentation updates to CmaEsSampler

### PR DESCRIPTION
**Pull request by turian**
## Motivation

Both myself and a colleague of mine thought we could not use CmaEsSampler for bools. But, we can.

## Description of the changes

Clarification that CmaEsSampler supports `suggest_discrete_uniform` and `suggest_int`

Related https://github.com/optuna/optuna/issues/2538

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@nzw0301 Could you review this PR if you have time?

File: optuna/samplers/_cmaes.py
Comment (HideakiImamura) [inline optuna/samplers/_cmaes.py:63]: I think it's more consistent to use the name of distributions.
```suggestion
    (However, DiscreteUniformDistribution and Int(Log)Distribution are supported.)
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR! I have a comment. Could you take a look?

File: optuna/samplers/_cmaes.py
Comment (nzw0301) [reply optuna/samplers/_cmaes.py:63]: @HideakiImamura 
Can we mention both `suggest` and `distribution` since a user samples a hyperparameter via either depending on interface: define-and-run or define-and-run?

Comment (turian) [reply optuna/samplers/_cmaes.py:63]: I will wait for an answer from @HideakiImamura

As a user, I am more familiar with the suggest_ interface.

Comment (turian) [reply optuna/samplers/_cmaes.py:63]: We could add both, if you like.


However, DiscreteUniformDistribution (`suggest_discrete_uniform`) and Int(Log)Distribution (`suggest_int`) ...

Comment (turian) [reply optuna/samplers/_cmaes.py]: Please check the revision

**Review by HideakiImamura — APPROVED** [review]
LGTM!

File: optuna/samplers/_cmaes.py
Comment (HideakiImamura) [reply optuna/samplers/_cmaes.py]: Thanks for the update. The current version looks good to me.

File: optuna/samplers/_cmaes.py
Comment (nzw0301) [inline optuna/samplers/_cmaes.py:64]: To generate links to related optuna pages, how about this?

```suggestion
    However, :class:`~optuna.distributions.DiscreteUniformDistribution`
    (:func:`~optuna.trial.Trial.suggest_discrete_uniform`) and
    Int(Log)Distribution (:func:`~optuna.trial.Trial.suggest_int`) are supported.
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]
Thank you for your suggestion! I have one comment. Could you look at my comment?

File: optuna/samplers/_cmaes.py
Comment (nzw0301) [reply optuna/samplers/_cmaes.py]: Thank you for your swift action!

File: optuna/samplers/_cmaes.py
Comment (turian) [reply optuna/samplers/_cmaes.py]: Good idea, pushed.

Comment (nzw0301) [reply optuna/samplers/_cmaes.py]: Thanks!

**Review by nzw0301 — APPROVED** [review]
LGTM! I'll merge this pull request after CI passes. Thank you for improving the document when you prepare two papers. I'm looking forward to reading them :)

**Comment by turian** [comment]
@nzw0301 yes of course. Could I jostle for some discussion on this issue? https://github.com/optuna/optuna/issues/1447

It would be very cool

**Comment by nzw0301** [comment]
Sure, I'll follow the discussion of #1447 and leave comments on #1447 to discuss with you.

**Pull request merged**
